packages/babel-preset-es2015/src/index.js

```diff
@@ -1,4 +1,27 @@
-module.exports = function(context, opts) {
+/**
+ * This file is a bit of a mess. If you're looking at it as a reference for how to write a preset,
+ * I'd recommend looking only at `function preset(){}` and ignoring the rest, unless your new preset
+ * really needs to work on babel-core < 6.13.x, which is unlikely.
+ */
+
+/**
+ * This preset was originally an object, before function-based configurable presets were introduced.
+ * For backward-compatibility with anything that may have been loading this preset and expecting
+ * it to be a simple Babel config object, we maintain the old config here.
+ */
+module.exports = preset({});
+
+// For backward compatibility with babel-core < v6.13.x, we use the 'buildPreset' property
+// of the preset object for the preset creation function.
+Object.defineProperty(module.exports, "buildPreset", {
+  configurable: true,
+  writable: true,
+  enumerable: false,
```

Comment:
Probably worth noting why this has to be non-enumerable, to avoid triggering the check for unknown options